### PR TITLE
Function and Pattern matching

### PR DESCRIPTION
**Pull request by WanjikuMac**
@zacck added exercise on Function and Pattern Matching

### CODE REVIEW

File: Fn_Pattern_Matching/Week_3_Exercise.md  [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Question 2
+-----------
+sum = fn(a,b,c) -> a+b+c end
+#Function<18.99386804/3 in :erl_eval.expr/5>
+sum.(1,2,3)
+6
+
+Question 3
+-----------
+pair_tuple_to_list = fn({abcd, efgh}) -> ([abcd, efgh]) end
```
Comment (zacck-zz) [inline Fn_Pattern_Matching/Week_3_Exercise.md ]: You can pattern match items with just a name so this
```
pair_tuple_to_list = fn({abcd, efgh}) -> ([abcd, efgh]) end
```
could also just be 
```
pair_tuple_to_list = fn({a, b}) -> ([a, b]) end
```

**Review by zacck-zz — COMMENTED** [review]
look at this too to see how to use issues and pull requests better https://blog.github.com/2016-02-17-issue-and-pull-request-templates/

File: Fn_Pattern_Matching/Parameterized_Fn.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Question 1
+-----------
+prefix = fn name -> (fn other -> "#{name} " <> " #{other}" end) end
+#Function<6.99386804/1 in :erl_eval.expr/5>
+mrs = prefix.("Mrs")
+#Function<6.99386804/1 in :erl_eval.expr/5>
+mrs.("smith")
```
Comment (zacck-zz) [inline Fn_Pattern_Matching/Parameterized_Fn.md:7]: Looks good now  make one of your own examples

**Comment by WanjikuMac** [comment]
@zacck Exercise on Parameterized_Fn added.

**Comment by WanjikuMac** [comment]
@zacck Exercise on ``& Notation`` added

**Review by zacck-zz — APPROVED** [review]

**Comment by WanjikuMac** [comment]
@zacck my own example on ``Parameterized Functions`` added

**Pull request merged**